src/langchain_google_cloud_sql_pg/cloudsql_vectorstore.py

```diff
@@ -0,0 +1,262 @@
+            values_stmt = f" VALUES ('{id}','{content}','{embedding}'"
+            extra = metadata
+            for metadata_column in self.metadata_columns:
+                if metadata_column in metadata:
+                    values_stmt += f",'{metadata[metadata_column]}'"
+                    del extra[metadata_column]
+                else:
+                    values_stmt += ",null"
+
+            insert_stmt += (
+                f", {self.metadata_json_column})" if self.store_metadata else ")"
+            )
+            values_stmt += f",'{json.dumps(extra)}')" if self.store_metadata else ")"
+            query = insert_stmt + values_stmt
+            await self.engine._aexecute(query)
+
+        return ids
+
+    async def aadd_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        embeddings = self.embedding_service.embed_documents(list(texts))
+        ids = await self._aadd_embeddings(
+            texts, embeddings, metadatas=metadatas, ids=ids, **kwargs
+        )
+        return ids
+
+    async def aadd_documents(
+        self,
+        documents: List[Document],
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        texts = [doc.page_content for doc in documents]
+        metadatas = [doc.metadata for doc in documents]
+        ids = await self.aadd_texts(texts, metadatas=metadatas, ids=ids, **kwargs)
+        return ids
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        try:
+            loop = asyncio.get_running_loop()
+            # loop = asyncio.get_event_loop()
+            return loop.run_until_complete(self.aadd_texts(texts, metadatas, ids))
+        except RuntimeError:
+            return self.engine.run_as_sync(self.aadd_texts(texts, metadatas, ids))
+
+    def add_documents(
+        self,
+        documents: List[Document],
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        try:
+            loop = asyncio.get_running_loop()
+            # loop = asyncio.get_event_loop()
+            return loop.run_until_complete(self.aadd_documents(documents, ids))
+        except RuntimeError:
+            return self.engine.run_as_sync(self.aadd_documents(documents, ids))
+
+    async def adelete(
+        self,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> Optional[bool]:
+        if ids:
+            id_list = ", ".join([f"'{id}'" for id in ids])
+            query = (
+                f"DELETE FROM {self.table_name} WHERE {self.id_column} in ({id_list})"
+            )
+            await self.engine._aexecute(query)
+            return True
+        else:
+            return False
```

Comment:
nit: happy path is default path
```suggestion
        if not ids:
            return False
            
        id_list = ", ".join([f"'{id}'" for id in ids])
        query = (
            f"DELETE FROM {self.table_name} WHERE {self.id_column} in ({id_list})"
        )
        await self.engine._aexecute(query)
        return True
```